lib/unix.js

```diff
@@ -1,7 +1,7 @@
-var exec = require('child_process').exec;
+var execFile = require('child_process').execFile;
 
 module.exports = function (iface, callback) {
-    exec("ifconfig " + iface, function (err, out) {
+    execFile("ifconfig", [iface], function (err, out) {
```

Comment:
If `iface` isn't properly sanitized wouldn't this still be vulnerable?

`ifconfig eth0 down`

Reply:
@javabudd No. `execFile` does not invoke a command interpreter / shell (unless an options dict with `shell` is passed, but the `shell` option defaults to `false`). The command is not simply concatenated with spaces and then interpreted again. Your example would yield `ifconfig 'eth0 down'` which is not two arguments `eth0` and `down` but one argument `etc0 down` which ifconfig will not find an interface for.

The previous vulnerability was that exec indeed does interpret the command and you could have done things like `ifconfig eth0; rm -rf /`. This is properly solved.